File: Bamilo/CartDataManager.m
////
////  CartDataManager.m
////  Bamilo
////
////  Created by Narbeh Mirzaei on 4/9/17.
////  Copyright © 2017 Rocket Internet. All rights reserved.
////
//
//#import "CartDataManager.h"
//
//@implementation CartDataManager
//
//#pragma mark - Overrides
//static CartDataManager *instance;
//
//+ (instancetype)sharedInstance {
//    static dispatch_once_t onceToken;
//    dispatch_once(&onceToken, ^{
//        instance = [CartDataManager new];
//    });
//    
//    return instance;
//}
//
//-(void)addProductToCart:(id<DataServiceProtocol>)target simpleSku:(NSString *)simpleSku completion:(DataCompletion)completion {
//    NSDictionary *params = @{ @"quantity": @1, @"sku": simpleSku };
//    
//    [self.requestManager asyncPOST:target path:RI_API_ADD_ORDER params:params type:RequestExecutionTypeForeground completion:^(NSInteger statusCode, ResponseData *data, NSArray *errorMessages) {
//        [self processResponse:statusCode ofClass:[RICart class] forData:data errorMessages:errorMessages completion:completion];
//    }];
//}
//
//- (void)getUserCart:(id<DataServiceProtocol>)target completion:(DataCompletion)completion {
//    [self.requestManager asyncPOST:target path:RI_API_GET_CART_DATA params:nil type:RequestExecutionTypeForeground completion:^(NSInteger statusCode, ResponseData *data, NSArray *errorMessages) {
//        [self processResponse:statusCode ofClass:[RICart class] forData:data errorMessages:errorMessages completion:completion];
//    }];
//}
//@end

File: CartDataManager.swift
//
//  CartDataManager.swift
//  Bamilo
//
//  Created by Narbeh Mirzaei on 6/6/17.
//  Copyright © 2017 Rocket Internet. All rights reserved.
//

import Foundation

class CartDataManager: DataManagerSwift {
    static let sharedInstance = CartDataManager()
    
    func addProductToCart(_ target: DataServiceProtocol, simpleSku: String, completion:@escaping DataClosure) {
        let params = [
            "quantity" : "1",
            "sku" : simpleSku
        ]
        let queryString = "?sku=\(simpleSku)"
        CartDataManager.requestManager.async(.post, target: target, path: RI_API_ADD_ORDER + queryString, params: params, type: .foreground) { (responseType, data, errorMessages) in
            self.processResponse(responseType, aClass: RICart.self, data: data, errorMessages: errorMessages, completion: completion)
        }
    }
    
    func getUserCart(_ target: DataServiceProtocol, type: ApiRequestExecutionType, completion:@escaping DataClosure) {
        CartDataManager.requestManager.async(.post, target: target, path: RI_API_GET_CART_DATA, params: nil, type: type) { (responseType, data, errorMessages) in
            self.processResponse(responseType, aClass: RICart.self, data: data, errorMessages: errorMessages, completion: completion)
        }
    }
}
